devito/operator.py

```diff
@@ -118,10 +118,10 @@ def __init__(self, expressions, **kwargs):
         self.symbolic_data = [i for i in parameters if isinstance(i, SymbolicData)]
         dims = [i for i in parameters if isinstance(i, Dimension)]
         dims_parents = [d.parent for d in dims if d.is_Buffered]
-        self.dims = list(set(dims + dims_parents))
+        self.dims = filter_ordered(dims + dims_parents)
```

Comment:
Yeah, that entire derivation logic seems to have redundancies. I'll have a look in a separate branch.